### PR TITLE
first draft of expectation maximization code.

### PR DESCRIPTION
**Pull request by acoadmarmon**
Still working on this expectation maximization code (this is a functional MVP), but wanted to create this PR so we can iteratively refine this approach @ProfFan .

### CODE REVIEW

File: Sources/BeeTracking/ProbabilisticTracker.swift [hunk 1]
```diff
@@ -229,6 +230,140 @@ public func makeProbabilisticTracker<
     })
 }
 
+public struct ProbablisticTracker<Encoder: AppearanceModelEncoder, FG: GenerativeDensity, BG: GenerativeDensity> {
```
Comment (ProfFan) [inline Sources/BeeTracking/ProbabilisticTracker.swift:236]: put this in a separate file?

**Comment by ProfFan** [comment]
@acoadmarmon Could you fix the CI error and merge? I can then fix the PPCA code.

**Comment by acoadmarmon** [comment]
Ya sure I'll let you know once I've fixed that test! Shouldn't take much time.

**Comment by acoadmarmon** [comment]
@ProfFan the latest commit should fix the CI error. Tests run locally.

**Comment by acoadmarmon** [comment]
@ProfFan I don't have write access so I can't merge this PR myself.

**Comment by ProfFan** [comment]
Oh, I'll merge, thanks :)

**Pull request merged**